src/controller/endpoint_slice.py

```diff
@@ -0,0 +1,169 @@
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+"""nginx-ingress-integrator k8s endpoint slice controller."""
+
+
+import typing
+
+import kubernetes.client
+
+from controller.resource import ResourceController, _map_k8s_auth_exception
+from ingress_definition import IngressDefinition
+
+
+class EndpointSliceController(
+    ResourceController[kubernetes.client.V1EndpointSlice]
+):  # pylint: disable=inherit-non-class
+    """Kubernetes EndpointSlice resource controller."""
+
+    def __init__(self, namespace: str, labels: typing.Dict[str, str]) -> None:
+        """Initialize the EndpointSliceController.
+
+        Args:
+            namespace: Kubernetes namespace.
+            labels: Label to be added to created resources.
+        """
+        self._ns = namespace
+        self._labels = labels
+        self._client = kubernetes.client.DiscoveryV1Api()
+        self._beta_client = kubernetes.client.DiscoveryV1beta1Api()  # pylint: disable=no-member
+
+    @property
+    def _name(self) -> str:
+        """Returns "endpoint slice"."""
+        return "endpoint slice"
+
+    @property
+    def _namespace(self) -> str:
+        """Returns the kubernetes namespace.
+
+        Returns:
+            The namespace.
+        """
+        return self._ns
+
+    @_map_k8s_auth_exception
+    def _gen_resource_from_definition(
+        self, definition: IngressDefinition
+    ) -> kubernetes.client.V1EndpointSlice:
+        """Generate a V1EndpointSlice resource from ingress definition.
+
+        Args:
+            definition: Ingress definition to use for generating the V1EndpointSlice resource.
+
+        Returns:
+            The generated V1EndpointSlice resource.
+        """
+        address_type = definition.upstream_endpoint_type
+        return kubernetes.client.V1EndpointSlice(
+            api_version="discovery.k8s.io/v1",
+            kind="EndpointSlice",
+            metadata=kubernetes.client.V1ObjectMeta(
+                name=definition.k8s_endpoint_slice_name,
+                namespace=definition.service_namespace,
+                labels={
+                    **self._labels,
```

Comment:
Will the "app.juju.is/created-by=nginx-ingress-integrator
                  nginx-ingress-integrator.charm.juju.is/managed-by=nginx-ingress-integrator" labels be included in here?

Reply:
yes, all those labels will be included in there